### ISSUE TITLE
Can't compile after ./configure --enable-log-bt=lvl

### ISSUE THREAD

**Issue by johnmeneghini** [spdk/spdk#388]
Configuring spdk with the following options fails to compile

```
 ./configure --enable-asan --enable-ubsan --enable-werror --enable-log-bt=lvl --enable-debug --enable-coverage 
```

## Expected Behavior

Any combination of ./configure flags should compile, link and run.

## Current Behavior

Compiling lib/log/log.c fails to compile with the above options. 

## Possible Solution

Don't configure --enable-log-bt=lvl 

## Steps to Reproduce

```
ssan-rx2560-01:spdk(master) > ./configure --enable-asan --enable-ubsan --enable-werror --enable-log-bt=lvl --enable-debug --enable-coverage 
ssan-rx2560-01:spdk(master) > make 
...
  CC lib/jsonrpc/jsonrpc_server_tcp.o
  LIB libspdk_jsonrpc.a
  CC lib/log/log.o
log.c: In function ‘spdk_log_unwind_stack’:
<command-line>:0:24: error: ‘SPDK_LOG_lvl’ undeclared (first use in this function); did you mean ‘SPDK_LOG_H’?
log.c:77:14: note: in expansion of macro ‘SPDK_LOG_BACKTRACE_LVL’
  if (level > SPDK_LOG_BACKTRACE_LVL) {
              ^~~~~~~~~~~~~~~~~~~~~~
<command-line>:0:24: note: each undeclared identifier is reported only once for each function it appears in
log.c:77:14: note: in expansion of macro ‘SPDK_LOG_BACKTRACE_LVL’
  if (level > SPDK_LOG_BACKTRACE_LVL) {
              ^~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [/home/johnm/SPDK/spdk/mk/spdk.common.mk:265: log.o] Error 1
make[1]: *** [/home/johnm/SPDK/spdk/mk/spdk.subdirs.mk:35: log] Error 2
make: *** [/home/johnm/SPDK/spdk/mk/spdk.subdirs.mk:35: lib] Error 2
```

## Context (Environment including OS version, SPDK version, etc.)

```
ssan-rx2560-01:spdk(master) > git logg -10
* b326e5121 2018-07-31  (HEAD -> master, origin/master) bdev/virtio_blk: defer bdev destruct completion [ Changpeng Liu / dariuszx.stojaczyk@intel.com ]
* a1ec258ac 2018-07-30  Clean up some incorrect Copyright statements [ Ben Walker / benjamin.walker@intel.com ]
* e67338d46 2018-07-30  bdev/raid: Factor out add raid bdev config operation [ Jim Harris / shuhei.matsumoto.xt@hitachi.com ]
* eff04f8e4 2018-07-30  bdev/raid: Factor out delete raid bdev config operation [ Jim Harris / shuhei.matsumoto.xt@hitachi.com ]
* e923c66a0 2018-07-30  bdev/raid: Free all strings allocated by JSON decoder on all paths in RPC [ Jim Harris / shuhei.matsumoto.xt@hitachi.com ]
```
```
ssan-rx2560-01:spdk(master) > uname -r
4.15.17-200.fc26.x86_64
```

**Comment by jimharris** [comment]
I can reproduce this on my dev system.

**Comment by jimharris** [comment]
Was going to assign to @pawelwod, but he's not part of the SPDK group yet.  If he accepts my invitation, we'll assign this to him.  (Pawel, please accept my invitation!)

**Comment by jimharris** [comment]
Oh - this isn't really a bug.  It should be something like --enable-log-bt=ERROR

"lvl" is just the placeholder for the log level.  I'm going to keep this issue open though as a bug and reduce it to low priority.  We should modify the configure script to throw an error if an invalid enable-log-bt flag is selected.

**Comment by WenqianZong** [comment]
I test this issue with "--enable-log-bt=ERROR" and "--enable-log-bt=lvl".
following is logs:
 "--enable-log-bt=ERROR"
```
log.c:40:10: fatal error: libunwind.h: No such file or directory
 #include <libunwind.h>
          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [/home/zong/spdk/mk/spdk.common.mk:265: log.o] Error 1
make[1]: *** [/home/zong/spdk/mk/spdk.subdirs.mk:35: log] Error 2
make: *** [/home/zong/spdk/mk/spdk.subdirs.mk:35: lib] Error 2

```

**Comment by chenlo2x** [comment]
I have resolved this issue on fedora 25, It is not a bug and just did not install the libunwind package.
You can install it by one of below cmd.
dnf install libunwind-devel.x86_64
apt install libunwind-dev
When you make the spdk,  the --enable-log-bt parameter should be followed by ERROR or other Log lever, not the lvl.
By the way, more details about the libunwind, please refer to below website.
http://www.nongnu.org/libunwind/download.html

**Comment by chenlo2x** [comment]
@jimharris & @johnmeneghini 
I have resolved this issue on fedora 25, It is not a bug and just did not install the libunwind package.
You can install it by one of below cmd.
dnf install libunwind-devel.x86_64
apt install libunwind-dev
When you make the spdk, the --enable-log-bt parameter should be followed by ERROR or other Log lever, not the lvl.
By the way, more details about the libunwind, please refer to below website.
http://www.nongnu.org/libunwind/index.html

**Comment by pawelwod** [comment]
@chenlo2x  @WenqianZong guys, you know that there is script in `scripts/pkgdep.sh` that will install all dependencies for you, right?

@jimharris @johnmeneghini I will add this check.

BTW: Are there any plans to use something like autoconf instead our own crafted script?

**Comment by benlwalker** [comment]
I investigated moving to autoconf at one point and couldn't figure out how to do a number of things that we need to do in config/make. So I just made a script that mostly works like the autoconf tools instead. In general, since our build system only needs to support Linux and FreeBSD, and even within those only relatively modern versions, I think this strategy is a lot simpler. I'm open to switching if someone were to get autoconf working and supporting all of the options we need.
